### DISCUSSION TITLE
My Home Assistant wundergroundpws stopped working

### DISCUSSION THREAD

**Discussion by sinemaker** [cytech/Home-Assistant-wundergroundpws#181] [General]
I am not a sophisticated user of HA.  I have HAOS running on a rasberry pi since January 2023.  I have been using this wundergroundpws integration to access data from my personal weather station.  It worked until about a week ago.  Since then it stopped.  I don't know how to debug the problem.  I get this line in my home-assistant.log file:

2023-07-17 15:55:39.093 ERROR (MainThread) [custom_components.wundergroundpws.wunderground_data] Error fetching WUnderground data: TimeoutError()

I tried getting a new API Key -- did not fix problem.
When I enter this command into my own browser it returns current pws data.

https://api.weather.com/v2/pws/observations/current?stationId=KMEPEMAQ5&format=json&units=e&apiKey=MYKEY&numericPrecision=decimal

I deleted all my wundergroundpws files from my config folder and followed instructions to upgrade to the latest version of this integration, changing commands in my configuration.yaml and sensor.yaml (where I have segregated all my sensor yaml commands).
No change.
I have tried reloading HA, rebooting HAOS -- no change.

I used to have a tp-link Deco X20 router.   About a week ago I also installed a new pfsense router (switching the Deco to be WiFi Access) and it is possible that I have not set up rules correctly and it is intercepting my HA requests to wunderground.

Basically I just need some help to trouble-shoot to figure out where things are getting dropped.

Sorry for the long post -- I just am at a loss.

**Comment by sinemaker** [comment]
Here is additional information from my home-assistant.log file that may be relevant:

2023-07-17 15:55:25.625 ERROR (MainThread) [hass_nabucasa.remote] Connection problem to snitun server
2023-07-17 15:55:39.093 ERROR (MainThread) [custom_components.wundergroundpws.wunderground_data] Error fetching WUnderground data: TimeoutError()
2023-07-17 15:55:39.099 ERROR (MainThread) [homeassistant.setup] Error during setup of component wundergroundpws
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 288, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/wundergroundpws/__init__.py", line 104, in async_setup
    raise PlatformNotReady
homeassistant.exceptions.PlatformNotReady: None
2023-07-17 15:55:39.619 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform wundergroundpws.sensor: Unable to set up component.
2023-07-17 15:55:39.623 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform wundergroundpws.weather: Unable to set up component.

**Reply by norburban** [reply]
I just discovered that you need to reboot the hardware HA is running on.  Updates usually only restart HA, but the latest updates require a full system reboot.  To do that, go to Settings / System / Hardware / click the power icon at the upper right / select Advanced options / Reboot system.

**Comment by cytech** [comment]
closed as outdated
